plugins/out_s3/s3.c

```diff
@@ -1073,6 +1084,17 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         if (chunk) {
             s3_store_file_unlock(chunk);
             chunk->failures += 1;
+            strftime(create_time_string, 80, "%Y-%m-%d %H:%M:%S", localtime(&(chunk->create_time)));
+            if (chunk->failures <= ctx->ins->retry_limit) {
+                s3_retry_warn(ctx->ins,
+                             "failed to flush chunk tag=%s, create_time=%s"
+                             "retry issued: input=%s > output=%s (out_id=%d)",
+                             (char *) chunk->fsf->meta_buf, create_time_string, m_upload->input_name, ctx->ins->name, ctx->ins->id);
+            } else {
+                s3_retry_warn(ctx->ins,
+                             "chunk tag=%s, create_time=%s cannot be retried: input=%s > output=%s",
+                             (char *) chunk->fsf->meta_buf, create_time_string, m_upload->input_name, ctx->ins->name);
+            }
```

Comment:
@Claych The goal of creating an S3 warn function in my thinking was to make all of this logic here be in one function. This makes code more readable and concise.